claims/verifier.go

```diff
@@ -0,0 +1,83 @@
+package claims
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Verifier bundles a number Assertions
+type Verifier struct {
+	Assertions Assertions
+}
+
+// NewVerifier initializes a new verifier
+func NewVerifier() *Verifier {
+
+	return &Verifier{
+		Assertions: make(map[Claim][]Assertion),
+	}
+
+}
+
+// Add adds an Assertion for a given Claim
+func (v *Verifier) Add(c Claim, a Assertion) *Verifier {
```

Comment:
Don't feel strongly about it, so feel free to leave it for now. I noted it because I'm moving towards immutable encrypters/signers in the go-jose.v2 base package.